File: Java/src/main/java/QuestionBank/Solution501_600/Solution566.java
package QuestionBank.Solution501_600;

/**
 * @author windows
 */
public class Solution566 {
    public int[][] matrixReshape(int[][] mat, int r, int c) {
        if (r * c != mat.length * mat[0].length){
            return mat;
        }
        int[] temp = new int[r * c];
        int index = 0;
        for (int i = 0; i < mat.length; i++) {
            for (int j = 0; j < mat[i].length; j++) {
                temp[index++] = mat[i][j];
            }
        }
        int[][] res = new int[r][c];
        index = 0;
        for (int i = 0; i < r; i++) {
            for (int j = 0; j < c; j++) {
                res[i][j] = temp[index++];
            }
        }
        return res;
    }


    public static void main(String[] args) {
        Solution566 solution566 = new Solution566();
    }
}

File: C++/src/Solution501_600/Solution566.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution566.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

vector<vector<int>> Solution566::matrixReshape(vector<vector<int>> &mat, int r, int c) {
    int m = mat.size();
    int n = mat[0].size();
    if (m * n != r * c) return mat;
    vector<vector<int>> res(r, vector<int>(c));
    int row = 0, col = 0;
    for (int i = 0; i < m; ++i) {
        for (int j = 0; j < n; ++j) {
            res[row][col++] = mat[i][j];
            if (col == c){
                row++;
                col = 0;
            }
        }
    }
    return res;
}
